scene/gui/line_edit.cpp

```diff
@@ -934,6 +958,29 @@ void LineEdit::_notification(int p_what) {
 				update();
 			}
 		} break;
+		case Control::NOTIFICATION_DRAG_BEGIN: {
+			drag_action = true;
+		} break;
+		case Control::NOTIFICATION_DRAG_END: {
+			drag_action = false;
+			drag_caret_force_displayed = false;
+		} break;
+		case Control::NOTIFICATION_DROP_DATA_SUCCESS: {
+			if (selection.drag_attempt) {
+				selection.drag_attempt = false;
+				if (is_editable()) {
+					delete_text(selection.begin, selection.end);
+					deselect();
+				} else if (deselect_on_focus_loss_enabled) {
+					deselect();
+				}
+			}
+		} break;
+		case Control::NOTIFICATION_DROP_DATA_FAIL: {
+			if (selection.drag_attempt) {
+				selection.drag_attempt = false;
+			}
```

Comment:
The condition here is not necessary, you could just always set it to `false`. Although not sure which option is theoretically better for performance.